projects/driver_controls/src/ads1015.c

```diff
@@ -0,0 +1,157 @@
+#include "ads1015.h"
+
+#include <stdbool.h>
+#include <stdio.h>
+
+#include "i2c.h"
+#include "delay.h"
+#include "gpio_it.h"
+#include "ads1015_bitmasks.h"
+
+// Union definition to help process registers data
+typedef struct ADS1015Register {
+  uint8_t data[2];
+} ADS1015Register;
+
+// Since the ADS1015 stores data in two's complement, an int16_t will be used to hold
+// the raw readings
+typedef struct ADS1015Interrupt {
+  ADS1015Callback callback;
+  void *context;
+  int16_t reading;
+} ADS1015Interrupt;
+
+static I2CPort s_i2c_port;
+static ADS1015Channel s_current_channel;
+
+static ADS1015Interrupt s_interrupts[NUM_ADS1015_CHANNELS];
+
+static StatusCode prv_channel_valid(ADS1015Channel channel) {
```

Comment:
I get that you're trying to reduce the amount of duplicated code with this function.

However, I think this means that the `STATUS_CODE` will be from the check function and not from the caller. Since we have no way of storing the caller, the error code isn't very helpful at that point.

I think this came up in a PR before, but I'm not sure what the consensus was. @ckitagawa?